### PR TITLE
[PATCH v3] Sysinfo update to include crypto capabilities

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

File: example/sysinfo/odp_sysinfo.c [hunk 1]
```diff
@@ -25,52 +25,206 @@ static const char *support_level(odp_support_t support)
...
+
+static void print_cipher_capa(odp_cipher_alg_t cipher)
+{
+	int caps = odp_crypto_cipher_capability(cipher, NULL, 0);
+	int rc, i;
+
+	if (caps < 0)
+		return;
+
+	odp_crypto_cipher_capability_t capa[caps];
```
Comment (muvarov) [inline example/sysinfo/odp_sysinfo.c]: put declaration on top please

File: example/sysinfo/odp_sysinfo.c [hunk 2]
```diff
@@ -25,52 +25,206 @@ static const char *support_level(odp_support_t support)
...
+	if (auths.bit.chacha20_poly1305)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_CHACHA20_POLY1305));
+}
+
+static void print_cipher_capa(odp_cipher_alg_t cipher)
+{
+	int caps = odp_crypto_cipher_capability(cipher, NULL, 0);
+	int rc, i;
+
+	if (caps < 0)
```
Comment (muvarov) [inline example/sysinfo/odp_sysinfo.c]: <= 0

File: example/sysinfo/odp_sysinfo.c [hunk 3]
```diff
@@ -25,52 +25,206 @@ static const char *support_level(odp_support_t support)
...
+		       capa[i].key_len, capa[i].iv_len);
+}
+
+static void print_auth_capa(odp_auth_alg_t auth)
+{
+	int caps = odp_crypto_auth_capability(auth, NULL, 0);
+	int rc, i;
+
+	if (caps < 0)
+		return;
```
Comment (Bill-Fischofer-Linaro) [inline example/sysinfo/odp_sysinfo.c]: Similar to earlier comment, `if (caps <= 0)` here.

File: example/sysinfo/odp_sysinfo.c (hunk 1 above)
Comment (lumag) [reply example/sysinfo/odp_sysinfo.c]: Not possible, please reread the code.

File: example/sysinfo/odp_sysinfo.c (hunk 2 above)
Comment (lumag) [reply example/sysinfo/odp_sysinfo.c]: ack

File: example/sysinfo/odp_sysinfo.c (hunk 3 above)
Comment (lumag) [reply example/sysinfo/odp_sysinfo.c]: ack

File: example/sysinfo/odp_sysinfo.c (hunk 1 above)
Comment (muvarov) [reply example/sysinfo/odp_sysinfo.c]: ah, yea, agree.

**Comment by muvarov** [comment]
@Bill-Fischofer-Linaro I'm ok with changes. What do you think?

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**